core/mgmt/v1beta/metric.proto

```diff
@@ -68,110 +55,119 @@ message PipelineTriggerTableRecord {
 }
 
-// PipelineTriggerChartRecord contains pipeline trigger metrics, aggregated by
+// PipelineTriggerChartRecord represents a timeline of pipeline triggers. It
+// contains a collection of (timestamp, count) pairs that represent the total
+// pipeline triggers in a given time bucket.
 // pipeline ID and time frame.
 message PipelineTriggerChartRecord {
-  // Pipeline ID.
-  string pipeline_id = 1;
-  // Pipeline UUID.
-  string pipeline_uid = 2;
-  // Trigger mode.
-  Mode trigger_mode = 3;
-  // Final status.
-  Status status = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // This field will be present present when the information is grouped by pipeline.
+  optional string pipeline_id = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // 2 is reserved for the pipeline UUID.
+  reserved 2;
+  // 3 is reserved for the trigger mode. The server wasn't grouping results by this
+  // field.
+  reserved 3;
+  // 4 is reserved for the trigger status. The server wasn't grouping results
+  // by this field.
+  reserved 4;
   // Time buckets.
   repeated google.protobuf.Timestamp time_buckets = 5 [(google.api.field_behavior) = OUTPUT_ONLY];
   // Aggregated trigger count in each time bucket.
   repeated int32 trigger_counts = 6 [(google.api.field_behavior) = OUTPUT_ONLY];
-  // Total computation time duration in each time bucket.
-  repeated float compute_time_duration = 7 [(google.api.field_behavior) = OUTPUT_ONLY];
-  // Version for the triggered pipeline if it is a release pipeline.
-  string pipeline_release_id = 8 [(google.api.field_behavior) = OUTPUT_ONLY];
-  // Release UUID for the triggered pipeline if it is a release pipeline.
-  string pipeline_release_uid = 9 [(google.api.field_behavior) = OUTPUT_ONLY];
-}
-
-// ListPipelineTriggerRecordsRequest represents a request to list the triggers
-// of a pipeline.
-message ListPipelineTriggerRecordsRequest {
-  // The maximum number of triggers to return. If this parameter is unspecified,
-  // at most 100 pipelines will be returned. The cap value for this parameter is
-  // 1000 (i.e. any value above that will be coerced to 100).
-  optional int32 page_size = 1 [(google.api.field_behavior) = OPTIONAL];
-  // Page token.
-  optional string page_token = 2 [(google.api.field_behavior) = OPTIONAL];
-  // Filter can hold an [AIP-160](https://google.aip.dev/160)-compliant filter
-  // expression.
-  // - Example: `create_time>timestamp("2000-06-19T23:31:08.657Z")`.
-  optional string filter = 3 [(google.api.field_behavior) = OPTIONAL];
+  // 7 is reserved for the trigger execution duration.
+  reserved 7;
+  // 8 is reserved for the pipeline release ID. The server wasn't grouping
+  // results by this field.
+  reserved 8;
+  // 9 is reserved for the pipeline release UUID. The server wasn't grouping
+  // results by this field.
+  reserved 9;
+  // Trigger requester ID, e.g. `users/specialist-wombat`.
+  string requester = 10 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Comment:
@donch1989 please take these (requester, owner) into account for the namespace unification. I tried to make these fields as descriptive as possible. E.g. right now the dashboard isn't clear about the fact the trigger count of a pipeline ID takes into account all the triggers, regardless the requester.

Reply:
ℹ️ if we wnat to display the triggers regardless the requester (e.g. the total triggers in a pipeline), this field should be optional.

Reply:
Right now, the dashboard is more like a collection display of data for the requester. If we want to show the triggers regardless of the requester, this data should belong to the pipeline/model logging page. I think it is fine to make the requester required here.